### DISCUSSION TITLE
2 inverters 1 gateway only returnig single inverter rsults vs combined values

### DISCUSSION THREAD

**Discussion by hidefandy** [TypQxQ/Sigenergy-Home-Assistant-Integration#44] [Q&A]
I have 2 x 6kw Intverters Ip xxx.xxx.x.55 & xxx.xx.x.82 and 1 gateway Ip xxx.xxx.xxx.66. I get data for individual inverters on Server Port 502 and  (Slave) Address 1. so that works.

I can't get combined or any values at all using the gateway IP with  (Slave) Address 1 or 247 (Plant?).  (Slave) Address 247 does not work for seperate inverters ip either.

**Comment by TypQxQ** [comment]
Hi! I haven't tried the modbus integration with multiple inverters but from what I understand of the documentation, you connect to the inverter that acts as master in the plant. You should have one with slave id 1 and other with slave id 2.

**Reply by hidefandy** [reply]
The master unit IP only returns data for the one unit regardless of what slave ID I use. It a bit strange as the id 1 or 247 does seem to have any impact. 

Both work?

I’m thinking the slave  ID maybe hard coded somewhere vs referring to the secret file

**Reply by TypQxQ** [reply]
There is no way to access two slave id's trough the current integration. I am considering rewriting it in Python so it can generate the entireties dynamically and then it would be possible to add multiple slave devices. It would not solve your problem though I think. 
The two inverters should communicate with each other. Can you see them both inside the app or do you have separate login for them?

**Reply by hidefandy** [reply]
I can see both inverters in the app.

The mysigem app Home page shows combined data. I can drill down in devices tab too both units, 4 batteries and 4 PV arrays.

Detailed data is by individual components.

It would take a lot of work to query each component and summarise everything like the native app. 

For now, I am just using the mySigen web portal as it’s own dashboard but I can’t get it to remember the authentication.

**Reply by TypQxQ** [reply]
From what I understand of the documentation on 247 you should get a summary for both and on 1 and 2 for each.
